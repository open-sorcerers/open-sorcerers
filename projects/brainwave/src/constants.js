export const BP = "basePath"
export const BR = "brain"
export const CO = "content" // comes from cosmiconfig
export const CC = "config" // comes from cosmiconfig
export const CF = "configFile"
export const FI = "filepath"
export const FT = "fileTypes"
export const MC = "control"
export const NS = "namespace"
export const RT = "root"
export const GLOB = "glob"
export const TK = "telepathy"
export const ST = "stats"
export const DA = "data"
export const DR = "dryRun"
export const FC = "fileContent"
export const RP = "relativePath"

export const INIT = `init`
