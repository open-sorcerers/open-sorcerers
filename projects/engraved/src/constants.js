export const C = {
  n: "\n"
}

export const REGEX_HEX3 = /[a-f0-9]{3}/
export const REGEX_HEX6 = /[a-f0-9]{6}/
