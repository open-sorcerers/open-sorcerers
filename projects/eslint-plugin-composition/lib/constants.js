const PARENT = 'parent';
const OBJECT = 'object';
const PROPERTY = 'property';
const NAME = 'name';
const TYPE = 'type';
const PROPS = 'props';

module.exports = {
  NAME,
  OBJECT,
  PARENT,
  PROPERTY,
  PROPS,
  TYPE
};
