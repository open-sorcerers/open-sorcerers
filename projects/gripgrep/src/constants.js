export const KEYS = Object.freeze({
  CWD: "cwd",
  REGEX: "regex",
  GLOBS: "globs",
  STRING: "string"
})

export const NIL = `nil`
