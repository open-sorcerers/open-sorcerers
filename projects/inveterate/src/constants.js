export const DEFAULT_CONFIG = { scheduled: false }

export const EVERY = Object.freeze({
  MINUTE: "* * * * *",
  FIVE_MINUTES: "*/5 * * * *",
  FIFTEEN_MINUTES: "*/15 * * * *",
  THIRTY_MINUTES: "*/30 * * * *",
  HOUR: "0 * * * *",
  MONTH: "0 0 1 * *",
  WEEKDAY_START: "0 9 * * 1-5",
  WEEKDAY_END: "0 5 * * 1-5",
  NIGHT: "0 0 * * *",
  WEEKEND_START: "0 0 * * 0",
  WEEKEND_END: "23 0 * * 6"
})

