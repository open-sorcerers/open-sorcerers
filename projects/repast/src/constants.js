import { map, pipe, reduce, mergeRight } from "ramda"
/* eslint-disable string-literal/no-string-literal */

export const REGEXES = {
  startsWithParens: /^\(/,
  endsWithParens: /\)$/
}

export const TYPES = {
  HM_TYPE: "RepastData",
  HM_SIGNATURE: "RepastSignature"
}

export const CHARACTERS = Object.freeze({
  __of__: `âˆ‹`,
  newline: "\n",
  comment: "//",
  space: " ",
  openParens: "(",
  closeParens: ")",
  asterisk: "*",
  empty: "",
  space: " ",
  doubleColon: "::",
  equal: "="
})
export const C = CHARACTERS

export const KEYWORDS = Object.freeze({
  REPAST: "@repast"
})
export const K = KEYWORDS

const makeObjectFromStrings = pipe(
  reduce((xx, yy) => mergeRight(xx, { [yy]: yy }), {}),
  Object.freeze
)

// read !cat file.js | grep 'L\..'
export const LITERALS = makeObjectFromStrings([
  "Boolean",
  "Global",
  "Identifier",
  "String",
  "ast",
  "arity",
  "body",
  "globals",
  "input",
  "json",
  "loc",
  "name",
  "nil",
  "object",
  "output",
  "params",
  "program",
  "signature",
  "string",
  "type",
  "undefined",
  "utf8",
  "value"
])

export const L = LITERALS

export const FLAGS = makeObjectFromStrings([L.input, L.output, L.json])
export const nf = map(z => "--" + z)

export const PRIMITIVE_TYPES = makeObjectFromStrings([
  "Boolean",
  "String",
  "Number",
  "Void",
  "Object",
  "Array",
]);

export const GLOBALS = pipe(
  reduce((agg, thing) => mergeRight(agg, { [thing.name]: thing }), {})
)([
  Array,
  Boolean,
  Date,
  Error,
  Function,
  Number,
  Object,
  Promise,
  Proxy,
  RegExp,
  String,
  Symbol
])
