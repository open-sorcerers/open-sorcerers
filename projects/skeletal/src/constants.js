export const UNSET = `%UNSET%`

export const STRINGS = Object.freeze({
  // cli stuff
  debug: `debug`,
  force: `force`,
  init: `init`,
  namespace: `namespace`,
  pattern: `pattern`,
  silent: `silent`,
  threads: `threads`,
  verbose: `verbose`,
  // errors
  NO_CONFIG: `noConfig`
})

export const CLI_OPTIONS = Object.freeze({
  number: ["t"],
  string: ["n", "p", "I"],
  boolean: ["f", "w", "s", "d"],
  default: {
    [STRINGS.debug]: false,
    [STRINGS.force]: false,
    [STRINGS.init]: "",
    [STRINGS.namespace]: "skeletal",
    [STRINGS.silent]: false,
    [STRINGS.threads]: 10,
    [STRINGS.verbose]: false
  },
  alias: {
    [STRINGS.debug]: ["d"],
    [STRINGS.force]: ["f"],
    [STRINGS.init]: ["I"],
    [STRINGS.namespace]: ["n"],
    [STRINGS.pattern]: ["p"],
    [STRINGS.silent]: ["s"],
    [STRINGS.threads]: ["t"],
    [STRINGS.verbose]: ["w"]
  }
})
