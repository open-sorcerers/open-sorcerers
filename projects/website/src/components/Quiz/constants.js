export const UNSET = 'UNSET'

export const GOOD = '👍🏽'
export const BAD = '👎🏽'
export const CHOICE = '✏️'
