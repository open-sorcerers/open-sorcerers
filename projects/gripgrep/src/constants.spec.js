import { KEYS, NIL } from "./constants"

test("KEYS", () => {
  expect(KEYS).toMatchSnapshot()
})

test("NIL", () => {
  expect(NIL).toMatchSnapshot()
})
