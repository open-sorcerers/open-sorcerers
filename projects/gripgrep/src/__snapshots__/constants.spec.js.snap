// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KEYS 1`] = `
Object {
  "CWD": "cwd",
  "GLOBS": "globs",
  "REGEX": "regex",
  "STRING": "string",
}
`;

exports[`NIL 1`] = `"nil"`;
